### PR TITLE
Add CI 

### PR DESCRIPTION
**Pull request by sananand007**
## Description
Add Jenkins CI using Docker Image

## Root Cause
Building, Running and testing the repo Code using Jenkins and Docker

## Solution
Using OnPremise servers for Master and slave Jenkins Node [everything is LinuxOS for deployement]

## Testing
No New tests are added as this is a infrastructure change

## Final Checks
- [ ] Have you removed commented code ?
- [ ] Have you used gofmt to format your code ?
- [ ] You have added unit tests

### CODE REVIEW

File: build_steps.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	token=$(jq '.token' $KEY_PATH)
+    token="${token%\"}"
+    token="${token#\"}"
+	echo $token
+}
+
+function post_gitStatus_success() {
+	echo "Post status after build succeeds"
+	git_token=$(extract_token)
+	URL="https://api.GitHub.com/repos/sananand007/JBot/statuses/$GIT_COMMIT?access_token=$git_token"
```
Comment (weihesdlegend) [inline build_steps.sh:34]: do we need to update this URL?

File: build_steps.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        }" >> output.txt
+
+}
+
+function push_docker() {
+	# Push Docker image to Dockerhub
+	echo "Push Step"
+}
+
+
```
Comment (weihesdlegend) [inline build_steps.sh:52]: pls remove the extra lines

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+FROM golang:latest as unwindtest
+
+# Check the working directory
+RUN ls -altr
+#RUN addgroup -S appuser
```
Comment (weihesdlegend) [inline Dockerfile:5]: consider remove the comments

File: build_steps.sh (hunk 1 above)
Comment (sananand007) [reply build_steps.sh:34]: Not required

File: Dockerfile (hunk 3 above)
Comment (sananand007) [reply Dockerfile:5]: sure .. final changes . I will remove these

Comment (sananand007) [reply Dockerfile:5]: All changes are removed

**Review by weihesdlegend — APPROVED** [review]
Lgtm

**Comment by sananand007** [comment]
This change will add the build steps file in the **master** branch and help in detecting the build scripts from master to run docker and CI on Jenkins server

**Pull request merged**